fix(formatter): Count emoji sequences correctly

leaysgur opened the pull request:
Fixes #17271

It seems `U+FE0F` is not counted as 1 by char. Please see https://github.com/oxc-project/oxc/pull/17331/changes#diff-68ae34cbf2a12b1e38d6edaa09bda33717157a1c90311e2adcd8f060eed45223R572-R579 for explanation.

Dunqing reviewed (changes requested):
Please add a test in the fixtures; I want to see the final output

Dunqing reviewed (approved):
Nice finding

leaysgur committed:
fix(formatter): Count emoji sequences correctly (#17331)

Fixes #17271

It seems `U+FE0F` is not counted as 1 by char. Please see https://github.com/oxc-project/oxc/pull/17331/changes#diff-68ae34cbf2a12b1e38d6edaa09bda33717157a1c90311e2adcd8f060eed45223R572-R579 for explanation.
